File: nodejs/src/policy.ts
import { Decision } from "./types.js";

export const HIGH_RISK_ACTIONS = new Set(["db_mutation", "send_email", "file_delete"]);

export function defaultPolicy(
  action: string,
  _resource: string | null,
  _agentId: string
): Decision {
  if (HIGH_RISK_ACTIONS.has(action)) return Decision.REQUIRE_APPROVAL;
  if (action.startsWith("admin_")) return Decision.BLOCK;
  return Decision.ALLOW;
}

File: rust/src/policy.rs
use crate::models::Decision;
use std::collections::HashSet;
use std::sync::LazyLock;

pub static HIGH_RISK_ACTIONS: LazyLock<HashSet<&'static str>> = LazyLock::new(|| {
    HashSet::from(["db_mutation", "send_email", "file_delete"])
});

pub fn default_policy(action: &str, _resource: Option<&str>, _agent_id: &str) -> Decision {
    if HIGH_RISK_ACTIONS.contains(action) {
        return Decision::RequireApproval;
    }
    if action.starts_with("admin_") {
        return Decision::Block;
    }
    Decision::Allow
}
